sktime/benchmarking/benchmarks.py

```diff
@@ -10,6 +10,39 @@
 from sktime.utils.validation._dependencies import _check_soft_dependencies
 
 
+def coer_estimator_and_id(estimators, estimator_id=None):
+    """Coerce estimators to a dict with estimator_id as key and estimator as value.
+
+    Parameters
+    ----------
+    estimators : dict, list or BaseEstimator object
+        Estimator to coerce to a dict.
+    estimator_id : str, optional (default=None)
+        Identifier for estimator. If none given then uses estimator's class name.
+
+    Returns
+    -------
+    estimators : dict
+        Dict with estimator_id as key and estimator as value.
+    """
+    VERSION_ID = "-v1"
+    if isinstance(estimators, dict):
+        return estimators
+    elif isinstance(estimators, list):
```

Comment:
Same as above.

Side comment: Pydantic helps to validate these things on it's own just by adding decorators, but I don't think we're ready to start using it. May be this can be discussed in one of the Friday calls.

Reply:
I added a validation function to check for the estimators object. Pls kindly review them again. 

I think most of the validations in sktime is done natively. adding pydantic means we are proposing adding pydantic as a dev dependency, right? Most likely need to go through meeting(s) with other core developers if im not mistaken.

Reply:
Agreed. I love to use pydantic, but for sktime it'd be a super big effort for sure. That's why made it a side note.